### PR TITLE
Chore/refactor add tests for carousel

### PR DESCRIPTION
**Pull request by ashleysimpson**
## Description

Refactor the test organization for the carousel component

## Motivation and Context

To increase code coverage and test the carousel functionality more rigorously 

## How Has This Been Tested?

These are new tests so it will inherently be tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] I have included a vue-play example (if this is a new feature)

### CODE REVIEW

File: tests/client/components/carousel.spec.js [hunk 1]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+      wrapper.vm.pauseAutoplay();
+      expect(wrapper.vm.autoplayInterval).toBe(undefined);
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    // start on first slide and allow the autoplay to advance to the second slide
+    it.skip('should display the second slide after waiting the timeout duration with direction forward', () => {});
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: These seem more like functional tests to me. I'd like to build something to do UI tests, so maybe we can move these it blocks into a to-do list for functional tests?

File: tests/client/components/carousel.spec.js [hunk 2]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+
+    // start on the second slide and allow autoplay to advanced to the first slide
+    it.skip('should display the first slide after waiting the timeout duration with direction backward', () => {});
+
+    it.skip('should not allow the autoplay feature to advance on hover', () => {});
+
+    it.skip('should allow the autoplay feature to advance on hover when hover pause false', () => {});
 
-    expect(wrapper.vm.canAdvanceBackward).toBeFalsy();
+    it.skip('should show the first slide twice when looping is enabled with autoplay', () => {});
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: What does `show slide twice` mean? Once at the start and once at the end of the array?

File: tests/client/components/carousel.spec.js [hunk 3]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+    it.skip('should center image when image is less than container width and centerMode is true', () => {});
+  });
+
+  describe('Custom easing property', () => {
+    it.skip('should set easing to ease by default', () => {});
 
-    expect(wrapper).toMatchSnapshot();
+    it.skip('should set easing to the custom easing supplied', () => {});
+
+    // TODO: Should we be testing for valid CSS transition easing like the README suggests?
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: Yeah i think so

File: tests/client/components/carousel.spec.js [hunk 4]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
   });
 
-  it('should apply custom slides per page when responsive param provided', () => {
-    const wrapper = mount(Carousel, {
-      propsData: {
-        perPageCustom: [[0, 20]]
-      }
+  describe('Dynamic slide count', () => {
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: Not 100% here but I think we can also test tabIndex of generated slides here

File: tests/client/components/carousel.spec.js [hunk 5]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
-      },
-      slots: {
-        default: [Slide, Slide, Slide, Slide, Slide]
-      }
+  describe('Carousel pagination', () => {
+    it.skip('should render the pagination component when pagination is enabled', () => {});
+
+    it.skip('should not render the pagination component when pagination is disabled', () => {});
+
+    // TODO: It seems that the pagination component is heavily connected to the carousel, decouple this?
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: Yes decouple for sure, let's add that as a point to the refactor

File: tests/client/components/carousel.spec.js [hunk 6]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+
+    it.skip('should only have one page when number of slides is less than slides per page', () => {});
   });
 
-  it('should register 3 slides when 3 slides are added to the slots', done => {
-    const wrapper = mount(Carousel, {
-      slots: {
-        default: [Slide, Slide, Slide]
-      }
+  describe('Resistance co-efficient', () => {
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: Just making a note here to have a better name for this, it's internal only right?

File: tests/client/components/carousel.spec.js [hunk 7]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+        wrapper.vm.computeCarouselWidth();
+
+        expect(wrapper.vm.currentPage).toBe(1);
+
+        done();
+      });
     });
 
-    expect(wrapper.vm.slideCount).toBe(3);
+    it.skip('should advanced the page by only one slide when scroll per page is false', () => {});
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: This seems like functional test aswell

File: tests/client/components/carousel.spec.js [hunk 8]
```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
...
+
+    it.skip('should not change the current page when value is negative', () => {});
+
+    it.skip('should not change the current page when value is greater than pageCount', () => {});
+  });
+
+  describe('Events', () => {
+    it.skip('should emit a pageChange event with the page number on page change', () => {});
+
+    it.skip('should emit a slideClick event with the object?? blah on slide click', () => {});
```
Comment (quinnlangille) [inline tests/client/components/carousel.spec.js]: Not sure what this is supposed to say lol

**Comment by quinnlangille** [comment]
Looks good overall, the reorganization will definitely be a huge help with inventory. Left a few comments, CC @josephting  for visibility in case you're working on more tests

File: tests/client/components/carousel.spec.js (hunk 1 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: Moved to a file so we can keep track of them

File: tests/client/components/carousel.spec.js (hunk 2 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: Yeah something like that, moved to a todo for now anyway

File: tests/client/components/carousel.spec.js (hunk 3 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: Cool I'll add tests for this

File: tests/client/components/carousel.spec.js (hunk 4 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: Added tests for this

File: tests/client/components/carousel.spec.js (hunk 6 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: This is a prop you pass in, I tried to make it more clear but it's hard to describe this in a test...

File: tests/client/components/carousel.spec.js (hunk 8 above)
Comment (ashleysimpson) [reply tests/client/components/carousel.spec.js]: Yeah I was confused about the event but I wanted to be sure to test it :1st_place_medal:

**Pull request merged**